packages/plugin-heading/Heading.ts

```diff
@@ -17,4 +55,29 @@ export class Heading extends JWPlugin {
             return [context, parsingMap];
         }
     }
+    /**
+     * Change the formatting of the nodes in given range to Heading.
+     *
+     * @param params
+     */
+    applyParagraph(params: ParagraphParams): void {
+        let heading = params.nodeCreator(); // todo: make sure to keep attributes
+        if (heading.is(HeadingNode)) {
+            const range = params.range || this.editor.vDocument.selection.range;
+            let nodesToConvert: VNode[];
+            if (range.isCollapsed()) {
+                nodesToConvert = [range.start.parent];
+            } else {
+                const selectedLeaves = range.selectedNodes(node => !node.children.length);
+                nodesToConvert = distinct(
+                    selectedLeaves.map(leaf => (leaf.atomic ? leaf.parent : leaf)),
+                );
```

Comment:
It's not the first time you have to do this construction. I have a hard time finding a good name for it but I think we will have to. I'm thinking something like `closestNonAtomic(selectedLeaves)` but it's not pretty. 🆘